### ISSUE TITLE
Use px as the default font size unit of measurement

### ISSUE THREAD

**Issue by mattpap** [bokeh/bokeh#6840]
Currently we use `pt` almost exclusively. However, `pt` is quite meaningless in the context of display on a screen. Given that everything else in bokeh in screen coordinates is given in pixels, I think we should apply the same to font sizes.

**Comment by bryevdv** [comment]
How is `px` interpreted on HiDPI vs non?

**Comment by mattpap** [comment]
> How is px interpreted on HiDPI vs non?

It's complicated, due to how "px" unit is defined and that it doesn't usually stand for actual pixels. However, that's not the point here. Whatever works for `pt` will work for `px`, just relative sizes will be different if not adjusted.

The point, however, is about those relative sizes. Suppose I have a circle of size `12`,  so `12px`. How many `pt` does it take to make text of equal height (assuming aspect ratio of 1 for simplicity)? Up until yesterday I would trial and error until I would find appropriate size. Now I know there's a conversion formula for this (`1pt` is `1/72` of `1in` and `1in` is `96px`). It would be much simpler if we used one unit of measurement. What would be affected is defaults (`TextProps.text_font_size`) and examples. Also, not using `pt` (and `in`, `mm`, `cm`) when targeting screens is a W3C recommendation. They recommend using `px` or `em` (or `rem`) (if you want to inherit / have scalable font size).

`em` and `rem` could be also useful, but only when also non-text glyphs would allow scaling as well.

**Comment by bryevdv** [comment]
I'm definitely not opposed to the idea, I just want to make sure `px` uses the "reported" pixel resolution on HiDPI displays, and not the actual pixel resolution (otherwise text would be smaller than intended)

**Comment by mattpap** [comment]
I made a test and used chrome's device emulation where I can set custom resolution and pixel density (here 1920x1080, DP=2.5, but I tested various configurations):
![image](https://user-images.githubusercontent.com/27475/29943543-1a2dc824-8e9a-11e7-8fb2-1fcef8c5f3c0.png)
On the left are text font sizes in `pt` and on the right in `px`, computed according to `72/96` ratio. Additionally there are circles with sizes corresponding to `px`.

Note that the image is quite ugly (I presume no aliasing), but this is how things look like in emulation mode.
